SimPEG/Examples/sphereElectrostatic_example.py

```diff
@@ -0,0 +1,775 @@
+from scipy.constants import epsilon_0
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import numpy as np
+from SimPEG.Utils import ndgrid, mkvc
+
+'''
+Authors: Thibaut Astic, Lindsey Heagy, Sanna Tyrvainen, Ronghua Peng
+Date: December 2015
+
+This code defines function to resolve analytically the electrostatic sphere problem.
+We first define a problem configuration, with a conductive or resistive sphere in a
+wholespace background.
+We then calculate the potential, then the electric field, then the current density and
+finally the charges accumulation.
+
+Several plotting functions are defined for data visualisation.
+
+
+'''
+
+# Plot options
+ftsize_title = 18      #font size for titles
+ftsize_axis  = 14      #font size for axis ticks
+ftsize_label = 14      #font size for axis labels
+
+# Radius function, useful sigma ratio, and log scale converter
+r  = lambda x,y,z: np.sqrt(x**2.+y**2.+z**2.)
+sigf = lambda sig0,sig1: (sig1-sig0)/(sig1+2.*sig0)
+
+#tools to convert log conductivity in conductivity
+def conductivity_log_wrapper(log_sig0,log_sig1):
+    sig0 = 10.**log_sig0
+    sig1 = 10.**log_sig1
+    
+    return sig0,sig1
+
+# Examples
+#Plot the configuration. Label=False is used to generate a general case figure
+def get_Setup(XYZ,sig0,sig1,R,E0,ax,label,colorsphere):
+    '''
+    XYZ: ndgrid
+    sig0: conductivity of the background
+    sig1: conductivity of the sphere
+    R: radius of the sphere
+    E0: Amplitude of the uniform electrostatic field
+    ax: ax where to plot the configuration
+    label: True: plot real values, False: plot general case
+    colorsphere: color of the sphere, format [x,x,x]
+    '''
+
+    xplt = np.linspace(-R, R, num=100)
+    xr,yr,zr = np.unique(XYZ[:,0]),np.unique(XYZ[:,1]),np.unique(XYZ[:,2])
+    dx = xr[1]-xr[0]
+    top = np.sqrt(R**2-xplt**2)
```

Comment:
we can replace this with a circle map (`Maps.CircleMap`) if instead of using line 53 as is, we use a SimPEG Mesh (`Mesh.TensorMesh`)